EarlGrey/Core/GREYKeyboard.m

```diff
@@ -36,9 +37,9 @@
 static GREYTapAction *gTapKeyAction;
 
 /**
- *  Flag set to @c YES when the keyboard is shown, @c NO when keyboard is hidden.
+ *  Flag set to @c true when the keyboard is shown, @c false when keyboard is hidden.
  */
-static BOOL gIsKeyboardShown = NO;
+static atomic_bool gIsKeyboardShown = false;
```

Comment:
TTBOMK, when I looked at this, `volatile` will only ensure that the compiler knows that other threads can modify the variable, thus stopping compiler optimizations, whereas `atomic` ensures that each operation on it is atomic in nature, thus providing greater thread-safety. Do you think there's another reason for using `volatile`?